pkgs/applications/graphics/freecad/default.nix

```diff
@@ -128,8 +128,14 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   # Their main() removes PYTHONPATH=, and we rely on it.
+  # Python >=3.11 support initializes using the "isolated" config
+  # way which ignores environment variables. As an alternative we
+  # could provide all our python packages with the `-P` cli option.
+  # As of 2023-10-26 HEAD of the main branch (as well as 0.21)
+  # still requires both of these modifications.
   preConfigure = ''
     sed '/putenv("PYTHONPATH/d' -i src/Main/MainGui.cpp
+    sed 's/PyConfig_InitIsolatedConfig/PyConfig_InitPythonConfig/' -i src/Base/Interpreter.cpp
```

Comment:
To me, personally, it looks like an upstreamable patch.

Reply:
I agree that talking to upstream might be a good idea. I just think this is a concious decision here as AFAIK this is a new feature of python 3.11.

Reply:
Since it is editing C++ code, can you please make a patch file instead of using sed?

Reply:
Would this be the correct patch?
```diff
diff --git a/src/Base/Interpreter.cpp b/src/Base/Interpreter.cpp
index f9853c1659..3a442daf58 100644
--- a/src/Base/Interpreter.cpp
+++ b/src/Base/Interpreter.cpp
@@ -556,7 +556,7 @@ void initInterpreter(int argc,char *argv[])
 {
     PyStatus status;
     PyConfig config;
-    PyConfig_InitIsolatedConfig(&config);
+    PyConfig_InitPythonConfig(&config);

     status = PyConfig_SetBytesArgv(&config, argc, argv);
     if (PyStatus_Exception(status)) {
diff --git a/src/Main/MainGui.cpp b/src/Main/MainGui.cpp
index 48ae847ef4..28813df383 100644
--- a/src/Main/MainGui.cpp
+++ b/src/Main/MainGui.cpp
@@ -112,17 +112,14 @@ int main( int argc, char ** argv )
     // See https://forum.freecad.org/viewtopic.php?f=18&t=20600
     // See Gui::Application::runApplication()
     putenv("LC_NUMERIC=C");
-    putenv("PYTHONPATH=");
 #elif defined(FC_OS_MACOSX)
     (void)QLocale::system();
-    putenv("PYTHONPATH=");
 #elif defined(__MINGW32__)
     const char* mingw_prefix = getenv("MINGW_PREFIX");
     const char* py_home = getenv("PYTHONHOME");
     if (!py_home && mingw_prefix)
         _putenv_s("PYTHONHOME", mingw_prefix);
 #else
-    _putenv("PYTHONPATH=");
     // https://forum.freecad.org/viewtopic.php?f=4&t=18288
     // https://forum.freecad.org/viewtopic.php?f=3&t=20515
     const char* fc_py_home = getenv("FC_PYTHONHOME");
```
For some reason it does not work when I overlay it in my flake based setup, but that might as well be just me messing up the overlay itself, as the eval error is the same as without the overlay :thinking:

Reply:
Well, it should be as easy as putting the file in `patches` list.

Reply:
I've updated the PR to contain a patch and tested it. Still works on x86_64-linux.